app/(dashboard)/dashboard/requests/[id]/components/ProductList.tsx

```
@@ -201,175 +199,147 @@ export default function ProductList({ requestId }: ProductListProps) {
-      // Revert local Zustand store state
-      if (context?.deletedProduct) {
-        // If we have the specific product that was deleted, add it back to the current store state
-        const currentProductsFromStore = useRequestDetailStore.getState().products;
-        const revertedProducts = [...currentProductsFromStore, context.deletedProduct as Product];
-        setProducts(revertedProducts);
-      } else if (context?.prevProducts) {
-        // Otherwise, if we have the snapshot of the entire product list before deletion, use that
-        setProducts(context.prevProducts as Product[]);
+        return { previousData, prevProducts, deletedProduct };
+      },
+      onError: (error, { /* id of product being deleted */ }, context) => {
```

Biome (1.9.4)

[error] Unexpected empty object pattern.



(lint/correctness/noEmptyPattern)